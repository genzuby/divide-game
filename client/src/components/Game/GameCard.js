import React from 'react';
import styled from 'styled-components';
import { PADDING } from '../../styles/MainStyle';
import { IMAGE, TITLE, DESCTEXT, ICONBG } from '../../styles/ComponentStyle';
import { BASIC_COLORS } from '../../styles/styleConstant';

const { MAIN_COLOR_TAKEAWAY, MAIN_TEXT_COLOR_WHITE } = BASIC_COLORS;

function GameCard({ isYou, playerid, playerinput, lastsum, result }) {
  const cardCompes = isYou ? 'right' : 'left';
  return (
    <GAMECARDBODY compose={cardCompes}>
      <GAMECARDINFO gridcolumn={isYou ? '2' : '1'}>
        <PADDING padding="1em" />
        <ICONBG size="3.8em">
          <IMAGE
            height="3.6em"
            src={isYou ? '/assets/myface.svg' : '/assets/competitor.svg'}
            alt={isYou ? 'your happy face' : "competitor's angry face"}
          />
        </ICONBG>
        <PADDING padding="0.5em" />
        <DESCTEXT
          style={{
            textAlign: 'center',
            whiteSpace: 'nowrap',
            width: '60px',
            overflow: 'hidden',
            textOverflow: 'ellipsis',
          }}
        >
          {isYou ? 'You' : playerid}
        </DESCTEXT>
      </GAMECARDINFO>
      <GAMECARDINFO gridcolumn={isYou ? '1' : '2'}>
        <GAMEPOINT compose={cardCompes}>
          <TITLE style={{ textAlign: 'center' }}>{playerinput}</TITLE>
        </GAMEPOINT>
        <PADDING padding="0.5em" />
        <GAMEDESC compose={cardCompes}>
          <DESCTEXT>
            [({playerinput}+{lastsum})/3] = {result}
          </DESCTEXT>
        </GAMEDESC>
        <PADDING padding="0.2em" />
        <GAMEDESC compose={cardCompes}>
          <DESCTEXT>{result}</DESCTEXT>
        </GAMEDESC>
      </GAMECARDINFO>
    </GAMECARDBODY>
  );
}

const GAMECARDBODY = styled.div`
  display: grid;
  grid-template-columns: ${props =>
    props.compose === 'right' ? `65% 35%` : `35% 65%`};
  grid-template-rows: 1fr;
  background-color: ${props =>
    props.compose === 'right' ? MAIN_COLOR_TAKEAWAY : MAIN_TEXT_COLOR_WHITE};
  border: 1px solid
    ${props =>
      props.compose === 'right' ? MAIN_TEXT_COLOR_WHITE : MAIN_COLOR_TAKEAWAY};
  color: ${props =>
    props.compose === 'right' ? MAIN_TEXT_COLOR_WHITE : MAIN_COLOR_TAKEAWAY};
  align-self: ${props =>
    props.compose === 'right' ? 'flex-end' : 'flex-start'};
  height: 120px;
  width: 240px;
  border-radius: 2px;
  margin-bottom: 1em;
  right: 0;
`;

const GAMECARDINFO = styled.div`
  grid-column: ${props => props.gridcolumn};
  grid-row: 1;
  disply: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 0.5em;
  div {
    align-self: center;
  }
`;

const GAMEPOINT = styled.div`
  background-color: ${props =>
    props.compose === 'right' ? MAIN_TEXT_COLOR_WHITE : MAIN_COLOR_TAKEAWAY};
  color: ${props =>
    props.compose === 'right' ? MAIN_COLOR_TAKEAWAY : MAIN_TEXT_COLOR_WHITE};
  border: 1px solid
    ${props =>
      props.compose === 'right' ? MAIN_COLOR_TAKEAWAY : MAIN_TEXT_COLOR_WHITE};
  height: 50%;
  width: 100%;
  border-radius: 0.5em;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const GAMEDESC = styled.div`
  padding: 0 0.5em;
  display: flex;
  justify-content: center;
  border: 1px solid
    ${props =>
      props.compose === 'right' ? MAIN_TEXT_COLOR_WHITE : MAIN_COLOR_TAKEAWAY};
`;

export default GameCard;
