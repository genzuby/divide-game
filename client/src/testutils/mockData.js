export const loginInfo = {
  isAvailable: true,
  userId: 'tomato',
  userKey: '12345678',
};

export const playerList = [
  {
    isAvailable: true,
    _id: '5fa7270bda31c407a193853a',
    userId: 'genzuby',
    userKey: '67012659-cd81-4f9e-843f-580d7b212b2b',
    __v: 0,
  },
  {
    isAvailable: true,
    _id: '5fa7274f019aaa08d961f6d8',
    userId: 'hoddie',
    userKey: '0f6b6dd6-2a5b-4718-b397-ecc7378beb3f',
    __v: 0,
  },
  {
    isAvailable: true,
    _id: '5fa7cdfe70559d085699e100',
    userId: 'hyunjoo',
    userKey: '0ae42d19-6dc5-4d9d-b2ba-52a6929eddbb',
    __v: 0,
  },
  {
    isAvailable: true,
    _id: '5fa72720da31c407a193853b',
    userId: 'shin',
    userKey: '3538278a-02a6-403a-9020-0c022cb6031a',
    __v: 0,
  },
  {
    isAvailable: true,
    _id: '5fa7dab888dcad0a93d567af',
    userId: 'test1234',
    userKey: 'c02c79e4-6cbd-4592-b069-be7be7fe9d66',
    __v: 0,
  },
  {
    isAvailable: true,
    _id: '5fa7ce0270559d085699e101',
    userId: 'tomato',
    userKey: '34d37561-562d-425d-8265-3c5d340fa43c',
    __v: 0,
  },
];

export const gamecontext_login = {
  gameInfo: {
    myid: 'tomato',
  },
};
