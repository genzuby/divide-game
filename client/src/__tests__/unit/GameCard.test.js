import React from 'react';
import { mount } from 'enzyme';
import '../../testutils/setup';
import GameCard from '../../components/Game/GameCard';

describe('Render Content', () => {
  let wrapper;

  beforeEach(() => {
    const props = {
      isYou: true,
      playerid: 'tomato',
      playerinput: 1,
      lastsum: 2,
      result: 3,
    };
    wrapper = mount(<GameCard {...props} />);
  });

  describe('Should render without error', () => {
    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
