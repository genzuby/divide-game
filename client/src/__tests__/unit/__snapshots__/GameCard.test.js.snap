// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Content Should render without error renders correctly 1`] = `ReactWrapper {}`;
